packages/wallets/src/components/Base/WalletTextField/WalletTextField.tsx

```diff
@@ -7,19 +7,22 @@ import './WalletTextField.scss';
 export interface WalletTextFieldProps extends ComponentProps<'input'>, HelperMessageProps {
     defaultValue?: string;
     disabled?: boolean;
+    dropdownWidth?: string;
```

Comment:
styling should be left to .css as a rule o thumb, unless there is really solid justification to use props for layout concerns,

Reply:
This was added as the width passed from the WalletListCardDropdown is variable and changes depending on the input width. As the logic for this is done outside of css it was passed as a a prop, unsure if it can be done in css

```
useEffect(() => {
        const selectedWallet = wallets?.find(wallet => wallet.loginid === loginid);
        if (selectedWallet) {
            const selectedTextWidth = `${selectedWallet.currency} ${
                selectedWallet.wallet_currency_type === 'Demo' ? 'Demo ' : ''
            }Wallet`.length;
            setDropdownWidth(`${selectedTextWidth - 2}rem`);
        }
    }, [wallets, loginid]);
```

https://github.com/binary-com/deriv-app/assets/142860499/f8a9632b-19f6-45bf-9823-0c8b3584971d

Reply:
I agree with @wojciech-deriv here. I don't understand the reason why the logic of this was done outside of css 🤔

Reply:
I think the approach in the useEffect above is not correct for multiple reasons. If it was added there already, then we should change it as its just not even correct. 

- 1rem equal font height, not width, ever letter have variable width, so calculation is not consistent by any means, and likely to just be incorrect for some languages
- for non-english langauges, width is gonna be still generated against english version, as it uses different string than the one in .jsx
- if I change string in .jsx, I also need to change string in that useEffect, which I will be completely unaware of, 

If it needs to be dynamic - usually best way to go is to use flex-shrink / flex-grow and other flex parameters.

Reply:
@wojciech-deriv @nijil-deriv the logic for this behaviour is done in js as the section of the component which needs the variable width is the `input`. Input tags behaves differently from `div` or `p` where the dynamic behaviour is limited hence adjusting the width through css methods such as flex or fit content wouldn't work. Here are some references:
- https://stackoverflow.com/questions/3392493/adjust-width-of-input-field-to-its-input
- https://css-tricks.com/auto-growing-inputs-textareas/

To address your concerns the calculation is updated to not use rem and a const generateTitleText with translation is added to be used in both the useEffect and passed in the WalletDropdown component. Let me know if there are better approaches for this

Reply:
thanks, yes, make sense, didn't realisized its input ;-/